libwire/inc/Sniffer.hpp

```diff
@@ -1,29 +1,34 @@
 #ifndef Sniffer_hpp
 #define Sniffer_hpp
 
-#include <stdio.h>
 #include <iostream>
 #include <vector>
+#include <memory>
+
 #include "SnifferObserver.hpp"
 #include "NetInterfaceInfo.h"
 #include "PacketReader.hpp"
 
 namespace Net {
-    class Sniffer {
-        
-    public:
-        explicit Sniffer(PacketReader *packetReader);
-        ~Sniffer() = default;
-        int startListening(InterfaceInfo const &interface);
-        int stopListening();
-        
-    private:
-        std::vector <class SnifferObserver *> _observers;
-        void notifyObserver();
-
-    protected:
-        PacketReader *_packetReader;
-    };
+
+class Sniffer {
+
+public:
+    explicit Sniffer(std::unique_ptr<PacketReader> packetReader);
```

Comment:
Use unique ptr to declare the ownership on the pointer,
unique_ptr will also automaticly call delete (free) on packetReader when sniffer will be destroyed